schema.development.kf

```diff
@@ -539,19 +564,38 @@ procedure share_attribute($id uuid, $original_attribute_id uuid, $attribute_key
 // ACCEESS GRANTS ACTIONS
 
 @kgw(authn='true')
-procedure has_grants($id uuid) public view returns (granted bool) {
-    for $row in SELECT 1 FROM access_grants WHERE data_id = $id::TEXT AND ag_grantee = @caller COLLATE NOCASE LIMIT 1 {
+procedure has_locked_grants($id uuid) public view returns (has bool) {
+    for $row in SELECT 1 FROM access_grants WHERE data_id = $id::TEXT AND ag_owner = @caller COLLATE NOCASE AND locked_until >= @block_timestamp LIMIT 1 {
         return true;
     }
     return false;
 }
 
-@kgw(authn='true')
-procedure has_locked_grants($id uuid) public view returns (has bool) {
-    for $row in SELECT 1 FROM access_grants WHERE data_id = $id::TEXT AND ag_owner = @caller COLLATE NOCASE AND locked_until >= @block_timestamp LIMIT 1 {
-        return true;
+// This procedure is used in an oracle resolution to determine the status of a AG at the moment of adding it into idOS
+// It needs to be done at this moment because of later defining data can change (original credential deleted, owner wallet or profile removed etc.)
```

Comment:
We should leave the door open to be able to recover from when a writer does things (create duplicate, create AG) in the wrong order.

For now, let's keep things as they are, since the current behavior assumes that we first create the duplicate, and only then create the AG. When we want to make this more fool-proof, we can call this function at the top of `get_credential_shared`, re-check if the duplicate was created, and change the AG status to `valid`.